src/backend/gpopt/gpdbwrappers.cpp

```diff
@@ -2596,6 +2599,13 @@ gpdb::MDCacheNeedsReset(void)
 {
 	GP_WRAP_START;
 	{
+		if ((gpdb::GPDBTransactionIdIsValid(
```

Comment:
I think it was bad idea to move logic here, because gpdb::MDCacheNeedsReset function is called unconditionally while logic - is conditionally

Reply:
I don't think anything has changed much. before this logic, only `CMDCache::FInitialized()` was called, which was just checking `NULL != m_pcache`. but this place looks more logical.

Reply:
If you return true via your new logic, then register_mdcache_invalidation_callbacks function is not called.

Reply:
yes it was a bad idea